src/views/settings/components/itemTypes.js

```diff
@@ -0,0 +1,232 @@
+import React, { PropTypes, Component } from 'react';
+import { Button, FormControl, FormGroup, Col, Form, ControlLabel,
+    ListGroup, ListGroupItem,
+} from 'react-bootstrap/lib/';
+import './settings.scss';
+
+export default class ItemTypes extends Component {
+
+    static propTypes = {
+        groups: PropTypes.arrayOf(Object),
+        itemName: PropTypes.string,
+        currentItem: PropTypes.string,
+        fieldVisible: PropTypes.boolean,
+        fieldType: PropTypes.string,
+        fieldName: PropTypes.string,
+        popupName: PropTypes.string,
+
+        setActiveItem: PropTypes.func.isRequired,
+        handleItemNameChange: PropTypes.func.isRequired,
+        fetchItemTypes: PropTypes.func.isRequired,
+        postItemType: PropTypes.func.isRequired,
+        postFieldType: PropTypes.func.isRequired,
+        setFieldVisible: PropTypes.func.isRequired,
+        setFieldName: PropTypes.func.isRequired,
+        setFieldType: PropTypes.func.isRequired,
+        removeField: PropTypes.func.isRequired,
+        removeItem: PropTypes.func.isRequired,
+        setPopupName: PropTypes.func.isRequired,
+    };
+
+    componentWillMount() {
+        const { fetchItemTypes } = this.props;
+        fetchItemTypes();
+    }
+
+    onFieldTypeDropDown = (e) => {
+        const { setFieldType } = this.props;
+        const type = e.target.value;
+        console.log(`Field Type Selected: ${type}`);
+        setFieldType(type);
+    };
+
+    handleOnItemSelect = (item) => {
+        const { groups, setActiveItem, setFieldVisible, setPopupName } = this.props;
+
+        const itemID = item.target.value;
+        console.log(itemID);
+        setActiveItem(itemID);
+        setFieldVisible(true);
+
+        // get the group name
+        let name = null;
+        for (let i = 0; i < groups.length; i += 1) {
+            if (groups[i].id === itemID) {
+                name = groups[i].name;
+            }
+        }
+
+        setPopupName(name);
+    };
+    addItem = () => {
+        const { postItemType, itemName, fetchItemTypes } = this.props;
+
+        if (itemName !== '' && itemName !== undefined && itemName !== null) {
+            postItemType(itemName).then((response) => {
```

Comment:
get rid of the `then` and just call `postItemType(itemName);`. you're already calling `fetchItemTypes` in `postItemType`